documentation/docs/05-load.md

```diff
@@ -0,0 +1,334 @@
+// ---cut---
+/** @type {import('./$types').PageLoad} */
+export function load() {
+	return {
+		a: Promise.resolve('a'),
+		b: Promise.resolve('b'),
+		c: {
+			value: Promise.resolve('c')
+		}
+	};
+}
+```
+
+```svelte
+<script>
+	/** @type {import('./$types').PageData} */
+	export let data;
+
+	console.log(data.a); // 'a'
+	console.log(data.b); // 'b'
+	console.log(data.c.value); // `Promise {...}`
+</script>
+```
+
+### Errors
+
+If an error is thrown during `load`, the nearest [`+error.svelte`](/docs/routing#error) will be rendered. For _expected_ errors, use the `error` helper from `@sveltejs/kit` to specify the HTTP status code and an optional message:
+
+```js
+/// file: src/routes/admin/+layout.server.js
+// @filename: ambient.d.ts
+declare namespace App {
+	interface Locals {
+		user: {
+			name: string;
+			isAdmin: boolean;
+		}
+	}
+}
+
+// @filename: index.js
+// ---cut---
+import { error } from '@sveltejs/kit';
+
+/** @type {import('./$types').LayoutServerLoad} */
+export function load({ locals }) {
+	if (!locals.user) {
+		throw error(401, 'not logged in');
+	}
+
+	if (!locals.user.isAdmin) {
+		throw error(403, 'not an admin');
+	}
+}
+```
+
+If an _unexpected_ error is thrown, SvelteKit will invoke [`handleError`](/docs/hooks#handleerror) and treat it as a 500 Internal Server Error.
+
+> In development, stack traces for unexpected errors are visible as `$page.error.stack`. In production, stack traces are hidden.
+
+### Redirects
+
+To redirect users, use the `redirect` helper from `@sveltejs/kit` to specify the location to which they should be redirected alongside a `3xx` status code.
+
+```diff
+/// file: src/routes/admin/+layout.server.js
+-import { error } from '@sveltejs/kit';
++import { error, redirect } from '@sveltejs/kit';
+
+/** @type {import('./$types').LayoutLoad} */
+export function load({ locals }) {
+	if (!locals.user) {
+-		throw error(401, 'not logged in');
++		throw redirect(307, '/login');
+	}
+
+	if (!locals.user.isAdmin) {
+		throw error(403, 'not an admin');
+	}
+}
+```
+
+### Invalidation
+
+SvelteKit tracks the dependencies of each `load` function to avoid re-running it unnecessarily during navigation. For example, a `load` function in a root `+layout.js` doesn't need to re-run when you navigate from one page to another unless it references `url` or a member of `params` that changed since the last navigation.
+
+Using [`invalidate(url)`](/docs/modules#$app-navigation-invalidate), you can re-run any `load` functions that depend on the invalidated resource (either implicitly, via [`fetch`](#fetch)), or explicitly via [`depends`](#depends). You can also invalidate _all_ `load` functions by calling `invalidate()` without an argument.
+
+### Shared state
+
+In many server environments, a single instance of your app will serve multiple users. For that reason, per-request state must not be stored in shared variables outside your `load` functions, but should instead be stored in `event.locals`.
```

Comment:
Since this not only relates to requests, but also the general user state, I suggest to enhance the section a little:

```suggestion
In many server environments, a single instance of your app will serve multiple users. For that reason, per-request state must not be stored in shared variables outside your `load` functions, but should instead be stored in `event.locals`. Similarly, per-user state must not be stored in global variables, instead make use of `$page.data` (which contains the combined data of all `load` functions) or use Svelte's [context feature](https://svelte.dev/docs#run-time-svelte-setcontext) to create scoped state.
```